content/blog/en/building-dojo-app.md

```diff
@@ -0,0 +1,1124 @@
+											break;
+										case 'large':
+											icache.set(
+												'currentPrice',
+												price + 1
+											);
+											break;
+									}
+								}}
+							>
+								{{
+									label: 'Drink Size',
+								}}
+							</RadioGroup>
+						</div>
+					),
+				}}
+			</Card>
+			<Card>
+				{{
+					content: (
+						<div classes={[css.content]}>
+							<CheckboxGroup
+								name="addins"
+								options={[
+									{
+										value: 'milk',
+										label: 'Milk',
+									},
+									{
+										value: 'sugar',
+										label: 'Sugar',
+									},
+									{
+										value: 'stevia',
+										label: 'Stevia',
+									},
+									{
+										value: 'honey',
+										label: 'Honey',
+									},
+								]}
+								onValue={(values) => {
+									icache.set('addins', values);
+								}}
+							>
+								{{
+									label: 'Add Ins',
+								}}
+							</CheckboxGroup>
+						</div>
+					),
+				}}
+			</Card>
+			<Card>
+				{{
+					content: (
+						<div classes={[css.content]}>
+							<CheckboxGroup
+								name="flavors"
+								options={[
+									{
+										value: 'caramel',
+										label: 'Caramel',
+									},
+									{
+										value: 'vanilla',
+										label: 'Vanilla',
+									},
+									{
+										value: 'pumpkin',
+										label: 'Pumpkin',
+									},
+									{
+										value: 'almond',
+										label: 'Almond',
+									},
+								]}
+								onValue={(values) => {
+									icache.set('flavors', values);
+								}}
+							>
+								{{
+									label: 'Flavors',
+								}}
+							</CheckboxGroup>
+						</div>
+					),
+				}}
+			</Card>
+			<Card>
+				{{
+					content: (
+						<div classes={[css.content]}>
+							<CheckboxGroup
+								name="toppings"
+								options={[
+									{
+										value: 'cinnamon',
+										label: 'Cinnamon',
+									},
+									{
+										value: 'whipped cream',
+										label: 'Whipped Cream',
+									},
+									{
+										value: 'nuts',
+										label: 'Nuts',
+									},
+								]}
+								onValue={(values) => {
+									icache.set('toppings', values);
+								}}
+							>
+								{{
+									label: 'Toppings',
+								}}
+							</CheckboxGroup>
+						</div>
+					),
+				}}
+			</Card>
+			...
+		</div>
+	);
+});
+```
+
+This looks like there is more happening here than there really is. We basically have a number of `Card` widgets that wrap a `CheckboxGroup`, which provides a list of options as checkboxes. When you click on a checkbox, the `onValue` method is kicked off and it provides an array of checked `values`. We can use these `values` to update our internal state with `icache`.
+
+```ts
+// per CheckboxGroup
+icache.set('addins', values);
+icache.set('flavors', values);
+icache.set('toppings', values);
+```
+
+The one that varies a bit, is the option to choose the size of the Drink. This is because the Drink size will update the Drink price.
```

Comment:
```suggestion
The approacj that varies a bit is the option to choose the size of the Drink. This is because the Drink size will update the Drink price.
```